docs/concepts/typings.md

```diff
@@ -4,4 +4,78 @@ title: Type hints
 
 # Type hints
 
-...
+Type hints are a modern feature of Python available since 3.5 whose existence is heavily influenced by the features of type-safe languages such as Rust. To learn more about type hints in Python, see [Real Python’s Type hinting walkthrough](https://realpython.com/lessons/type-hinting/).
+
+When using Strawberry to build graphQL APIs, as was shown in [Schema basics](https://strawberry.rocks/docs/general/schema-basics), type hints are required within classes decorated by `@strawberry.type` & `@strawberry.input` and functions decorated by `@strawberry.field` & `strawberry.mutation`. These type hints are sourced as the keywords `str`, `int`, `float`, and from packages imported directly from the Python standard libraries `typing`, that has been available since Python 3.5, `datetime`, and `decimal`.
+
+## Mapping to graphQL types
+
+The complete mapping of the required type hints for the relevant graphQL types is as follows:
+
+| GraphQL         | Python            |
+| --------------- | ----------------- |
+| `ID`            | `strawberry.ID`   |
+| `String`        | `str`             |
+| `Integer`       | `int`             |
+| `Float`         | `float`           |
+| `Array` or `[]` | `typing.List`     |
+| `Union`         | `typing.Union`    |
+| `Nullable`      | `typing.Optional` |
+| `Date`          | `datetime.date`   |
+| `Decimal`       | `decimal.Decimal` |
+
+where `typing`, `datetime`, and `decimal` are all part of the Python standard library.
+
+The usefulness of using `typing.List` instead of the keyword `list` is so that the types held in the list can be notated. A list of integers is notated as `typing.List[int]` and not as `list[int]`. The only effective equivalence that can be understood by the machine is that `typing.List[typing.Any]` is equivalent to `list`.
```

Comment:
from python 3.10 you can use `list[str]` as an equivalent to `typing.List[str]` :)

I think we should make a not for that, or maybe use it as the default 

There's also `int | None` for optional types and `TypeA | TypeB` for union types